src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/RuntimeInformation.Windows.cs

```diff
@@ -70,5 +89,28 @@ private static Architecture Map(int processorArchitecture)
                     return Architecture.X86;
             }
         }
+
+        private static Architecture MapMachineConstant(ushort processMachine)
+        {
+            switch (processMachine)
+            {
+                case 0x01C0: // IMAGE_FILE_MACHINE_ARM
+                case 0x01C2: // IMAGE_FILE_MACHINE_THUMB
+                case 0x01C4: // IMAGE_FILE_MACHINE_ARMNT
+                    return Architecture.Arm;
+
+                case 0x8664: // IMAGE_FILE_MACHINE_AMD64
+                    return Architecture.X64;
+
+                case 0xAA64: // IMAGE_FILE_MACHINE_ARM64
+                case 0xA641: // IMAGE_FILE_MACHINE_ARM64EC
+                case 0xA64E: // IMAGE_FILE_MACHINE_ARM64X
+                    return Architecture.Arm64;
+
+                default: // IMAGE_FILE_MACHINE_UNKNOWN etc.
+                    Debug.Assert(processMachine != 0x014C /* IMAGE_FILE_MACHINE_I386 */, "Unidentified Architecture");
+                    return Architecture.X86;
```

Comment:
It would be better to return `ProcessArchitecture` for the unknown cases. It is more likely to be correct than x86.

Reply:
Would it be better to use `GetNativeSystemInfo` instead of `ProcessArchitecture` first? The `Map` function also has a similar fallback to the x86 one here, and it may be good to change it as well.

Reply:
I do not think it makes sense to call `GetNativeSystemInfo` here on systems that have `IsWow64Process2` API. There is no known cases where this would help.

We will want to delete the `GetNativeSystemInfo` path completely once the minimum supported Windows OS is Windows Server 2016+ / Windows 10, version 1511 that is guaranteed to have `IsWow64Process2`. It may be worth adding a comment about it.